File: server/internal/model/operation_log.go
package model

import "time"

// OperationLog 是后台操作日志模型。
type OperationLog struct {
	ID           uint      `gorm:"primaryKey" json:"id"`
	UserID       uint      `gorm:"not null;default:0;index" json:"user_id"`
	Username     string    `gorm:"size:64;not null;default:'';index" json:"username"`
	Method       string    `gorm:"size:10;not null;index" json:"method"`
	Path         string    `gorm:"size:255;not null;index" json:"path"`
	RoutePath    string    `gorm:"size:255;not null;default:'';index" json:"route_path"`
	Query        string    `gorm:"size:1000;not null;default:''" json:"query"`
	IP           string    `gorm:"column:ip;size:64;not null;default:''" json:"ip"`
	UserAgent    string    `gorm:"size:500;not null;default:''" json:"user_agent"`
	StatusCode   int       `gorm:"not null;default:0;index" json:"status_code"`
	LatencyMs    int64     `gorm:"not null;default:0" json:"latency_ms"`
	Success      bool      `gorm:"not null;default:true;index" json:"success"`
	ErrorMessage string    `gorm:"size:500;not null;default:''" json:"error_message"`
	CreatedAt    time.Time `json:"created_at"`
}

// TableName 固定操作日志表名。
func (OperationLog) TableName() string {
	return "sys_operation_log"
}

File: admin/src/modules/system/types/operation-log.ts
// OperationLogItem 类型定义。
export interface OperationLogItem {
  id: number
  user_id: number
  username: string
  method: string
  path: string
  route_path: string
  query: string
  ip: string
  user_agent: string
  status_code: number
  latency_ms: number
  success: boolean
  error_message: string
  created_at: string
}

// OperationLogListQuery 类型定义。
export interface OperationLogListQuery {
  page: number
  page_size: number
  username?: string
  method?: string
  path?: string
  success?: string
}

// OperationLogListResponse 类型定义。
export interface OperationLogListResponse {
  items: OperationLogItem[]
  total: number
  page: number
  page_size: number
}
